include/common/mpmc_queue.h

```diff
@@ -187,6 +187,8 @@ constexpr bool is_queue = internal::is_queue<T>();
 template <typename T>
 using requires_queue = std::enable_if_t<is_queue<T>, int>;
 
+template <typename Queue, typename T, typename ...>
+constexpr bool check_queue_type() { return std::is_same<Queue,T>::value;}
```

Comment:
Bad naming. Should be something like `is_queue_type`.
Why do we need the variadic argument?
Perhaps documentation would help here.

Reply:
Renamed. 
The variadic is necessary beacuse it will receive a parameter pack and only the first one will be checked.